### PR TITLE
Mul 1616 btc seg wit logic tx

### PR DESCRIPTION
**Pull request by PashaKlybik**
MUL-1616: Make Bitcoin segwit Tx
Serialize SegWit transaction
Sign SegWit transaction
Refactor testcase for Bitcoin SegWit
Refactor bitcoin stream

### CODE REVIEW

File: multy_core/src/bitcoin/bitcoin_transaction_segwit.cpp [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+    {
+        preview_inputs << *(static_cast<const BitcoinTransactionSegWitSource*>(source.get())->preview_transaction());
+    }
+    BitcoinDataStream preview_outputs;
+    for (const auto& destination : m_destinations)
+    {
+        preview_outputs << *destination;
+    }
+
+    BitcoinDataStream data;
```
Comment (Enmk) [inline multy_core/src/bitcoin/bitcoin_transaction_segwit.cpp]: `sequence_stream` ?

File: multy_core/src/bitcoin/bitcoin_transaction_segwit.cpp [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+    for (const auto& source : m_sources)
+    {
+        data << *source->sequence;
+    }
+
+
+    hash<256> hash_prevouts = do_hash<SHA2_DOUBLE, 256>(preview_inputs.get_content());
+    hash<256> hash_sequence = do_hash<SHA2_DOUBLE, 256>(data.get_content());
+    hash<256> hash_outpusts = do_hash<SHA2_DOUBLE, 256>(preview_outputs.get_content());
```
Comment (Enmk) [inline multy_core/src/bitcoin/bitcoin_transaction_segwit.cpp]: Why `hash_prevouts` is the hash of `preview_inputs`, and hash of `preview_outputs` is named so weirdly - `hash_outpusts` ?

File: multy_core/src/bitcoin/bitcoin_transaction_segwit.cpp [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+
+    for (auto& source : m_sources)
+    {
+        static_cast<BitcoinTransactionSegWitSource*>(
+                source.get())->sign(hash_prevouts, hash_sequence, hash_outpusts, *lock_time);
+    }
+}
+
+void BitcoinTransactionSegWit::verify() const
```
Comment (Enmk) [inline multy_core/src/bitcoin/bitcoin_transaction_segwit.cpp]: It looks like we are not going to have a change address feature for segwit transactions, which is bad.

Comment (PashaKlybik) [reply multy_core/src/bitcoin/bitcoin_transaction_segwit.cpp]: This logic will be implement in the future.

**Pull request merged**